src/main/java/utils/Storage.java

```diff
@@ -0,0 +1,137 @@
+package utils;
+
+import exceptions.DukeException;
+import tasks.Task;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Todo;
+
+import java.text.ParseException;
+import java.util.Scanner;
+import java.util.List;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Storage {
+    private List<Task> tasks;
+    private File file;
+    private String filePath;
+
+    /**
+     * Enable read and write access to retrieve and store tasks.
+     *
+     * @param filePath the local path to storage file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.file = new File(filePath);
+        this.tasks = new ArrayList<>();
+
+    }
+
+    /**
+     *  Takes the stored data and populates the task list.
+     *
+     * @return list of tasks
+     * @throws DukeException if local file not found or invalid date in saved file
+     */
+    public List<Task> readFromFile() throws DukeException {
+        Scanner sc;
+        try {
+            if (file.createNewFile()) {
```

Comment:
I like how you have outputs for all your steps; even for the creation or checking the existence of the 'dukeTest.txt' file as it really aids in debugging.
Just my opinion though, but I think the user does not need to know or see this.
Also, for the javadoc comment up top, it states that a DukeException is thrown if the local file is not found. I feel that it could be extrapolated on slightly as you are immediately creating a new file if it does not exist.
Do correct me if I am wrong on how it works though.